x-pack/plugins/ml/public/application/model_management/test_models/models/text_expansion/text_expansion_inference.ts

```diff
@@ -0,0 +1,177 @@
+      defaultMessage:
+        'Expand your search to include relevant terms in the results that are not present in the query.',
+    }),
+  ];
+
+  public queryText$ = new BehaviorSubject<string>('');
+  private queryResults: Record<string, number> = {};
+
+  constructor(
+    trainedModelsApi: ReturnType<typeof trainedModelsApiProvider>,
+    model: estypes.MlTrainedModelConfig,
+    inputType: INPUT_TYPE,
+    deploymentId: string
+  ) {
+    super(trainedModelsApi, model, inputType, deploymentId);
+
+    this.initialize(
+      [this.queryText$.pipe(map((questionText) => questionText !== ''))],
+      [this.queryText$]
+    );
+  }
+
+  protected async inferText() {
+    return this.runInfer<estypes.MlInferTrainedModelResponse>(
+      () => {},
+      (resp, inputText) => {
+        return {
+          response: parseResponse(
+            resp as unknown as MlInferTrainedModelResponse,
+            '',
+            this.queryResults
+          ),
+          rawResponse: resp,
+          inputText,
+        };
+      }
+    );
+  }
+
+  protected async inferIndex() {
+    const { docs } = await this.trainedModelsApi.trainedModelPipelineSimulate(this.getPipeline(), [
+      {
+        _source: {
+          text_field: this.getQueryText(),
+        },
+      },
+    ]);
+    this.queryResults = docs[0].doc?._source[this.inferenceType].predicted_value ?? {};
+
+    return this.runPipelineSimulate((doc) => {
+      return {
+        response: parseResponse(
+          { inference_results: [doc._source[this.inferenceType]] },
+          doc._source[this.getInputField()],
+          this.queryResults
+        ),
+        rawResponse: doc._source[this.inferenceType],
+        inputText: doc._source[this.getInputField()],
+      };
+    });
+  }
+
+  protected getProcessors() {
+    return this.getBasicProcessors();
+  }
+
+  public setQueryText(text: string) {
+    this.queryText$.next(text);
+  }
+
+  public getQueryText$() {
+    return this.queryText$.asObservable();
+  }
+
+  public getQueryText() {
+    return this.queryText$.getValue();
+  }
+
+  public getInputComponent(): JSX.Element | null {
+    const placeholder = i18n.translate(
+      'xpack.ml.trainedModels.testModelsFlyout.textExpansion.inputText',
+      {
+        defaultMessage: 'Enter a phrase to test',
+      }
+    );
+    return getTextExpansionInput(this, placeholder);
+  }
+
+  public getOutputComponent(): JSX.Element {
+    return getTextExpansionOutputComponent(this);
+  }
+}
+
+interface MlInferTrainedModelResponse {
+  inference_results: TextExpansionPredictedValue[];
+}
+
+interface TextExpansionPredictedValue {
+  predicted_value: Record<string, number>;
+}
+
+function parseResponse(
+  resp: MlInferTrainedModelResponse,
+  text: string,
+  queryResults: Record<string, number>
+): FormattedTextExpansionResponse {
+  const [{ predicted_value: predictedValue }] = resp.inference_results;
+
+  if (predictedValue === undefined) {
+    throw new Error('No results found');
```

Comment:
is it a user facing error?

Reply:
It is, updated in https://github.com/elastic/kibana/pull/159150/commits/faefba5fff6817c3b79e296a6749e1ed37bed2ef